Kannst Du Dich darum noch kümmern. Ansonsten bin ich mit dem Design höchst zufrieden. Good job!

db/db.h:26:65: error: 'UINT64' was not declared in this scope
   std::vector<std::tuple<std::string,std::string,double,double, UINT64>>  getEncounters(std::string puplicKeyContactPerson);

db/db.cpp:15:74: error: template argument 2 is invalid
db/db.cpp:17:10: error: request for member 'clear' in 'result', which is of non-class type 'int'
   result.clear();

db/db.cpp:81:40: error: request for member 'at' in '_encounters', which is of non-class type 'int'
     outfile << std::get<0>(_encounters.at(i));

db/db.cpp: In member function 'std::vector<std::__cxx11::basic_string<char> > DB::splitString(std::__cxx11::string, std::__cxx11::string)':
db/db.cpp:107:72: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   while ((endIndex = _stringToBeSplitted.find(_delimeter, startIndex)) < _stringToBeSplitted.size())

--- a/db/db.cpp
+++ b/db/db.cpp
@@ -5,16 +5,15 @@
 DB::DB(std::string _path, std::string _fileEnding) :
   m_path(_path),
   m_fileEnding(_fileEnding)
-{
-}
+{}
+
 //############################################################################################################
 DB::~DB() {}
+
 //############################################################################################################
 std::vector<std::tuple<std::string, std::string, double, double, UINT64>>  DB::getEncounters(std::string _puplicKeyContactPerson)
 {
   std::vector<std::tuple<std::string, std::string, double, double, UINT64>> result;
-
-  result.clear();
 
   std::string path = getFilenameOfContact(_puplicKeyContactPerson);
 
@@ -27,10 +26,6 @@ std::vector<std::tuple<std::string, std::string, double, double, UINT64>>  DB::g
   {
 	std::string str;
 
-	infile.seekg(0, std::ios::end);
-	str.reserve(infile.tellg());
-	infile.seekg(0, std::ios::beg);
-
 	str.assign((std::istreambuf_iterator<char>(infile)),
 	std::istreambuf_iterator<char>());
 
@@ -41,7 +36,8 @@ std::vector<std::tuple<std::string, std::string, double, double, UINT64>>  DB::g
 
 	// Fill result
 	std::tuple<std::string, std::string, double, double, UINT64> temp;
-	for (int i = 0; i < lines.size(); i++)
+
+	for (size_t i = 0; i < lines.size(); i++)
 	{ 
 		//Split line into string vector
 		std::istringstream iss(lines[i]);
@@ -51,8 +47,8 @@ std::vector<std::tuple<std::string, std::string, double, double, UINT64>>  DB::g
 		//Set values into tuple
 		std::get<0>(temp) = tokens[0]; // contact
 		std::get<1>(temp) = tokens[1]; // infected
-		std::get<2>(temp) = ::atof(tokens[2].c_str());	// longitude
-		std::get<3>(temp) = ::atof(tokens[3].c_str());		// altitude
+		std::get<2>(temp) = ::atof(tokens[2].c_str());	// latitude
+		std::get<3>(temp) = ::atof(tokens[3].c_str());		// longitude
 		std::get<4>(temp) = std::stoull(tokens[4].c_str());	// timestamp
 
 		//Add tuple to result vector
@@ -69,27 +65,28 @@ std::vector<std::tuple<std::string, std::string, double, double, UINT64>>  DB::g
   mtx.unlock();
   return result;
 }
+
 //############################################################################################################
 void DB::submitEncounters(std::vector<std::tuple<std::string, std::string, double, double, UINT64>> _encounters)
 {
-  for(unsigned int i = 0; i < _encounters.size(); i++){
+  for(size_t i = 0; i < _encounters.size(); i++){
     std::string path = getFilenameOfContact(std::get<0>(_encounters.at(i)));
     std::ofstream outfile;
     mtx.lock();
     outfile.open(path.c_str(), std::ios_base::app);
-	// Write puplicKeyContactPerson
+    // Write puplicKeyContactPerson
     outfile << std::get<0>(_encounters.at(i));
     outfile << " ";
-	// Write puplicKeyInfectedPerson
+    // Write puplicKeyInfectedPerson
     outfile << std::get<1>(_encounters.at(i));
     outfile << " ";
-	// Write longitude
+    // Write latitude
     outfile << std::get<2>(_encounters.at(i));
 	outfile << " ";
-	// Write altitude
+    // Write longitude
 	outfile << std::get<3>(_encounters.at(i));
 	outfile << " ";
-	// Write timestamp
+    // Write timestamp
 	outfile << std::get<4>(_encounters.at(i));
 
     outfile << std::endl;
@@ -98,12 +95,13 @@ void DB::submitEncounters(std::vector<std::tuple<std::string, std::string, doubl
   }
   return;
 }
+
 //############################################################################################################
 std::vector<std::string> DB::splitString(std::string _stringToBeSplitted, std::string _delimeter)
 {
   std::vector<std::string> splittedString;
-  int startIndex = 0;
-  int  endIndex = 0;
+  size_t startIndex = 0;
+  size_t endIndex = 0;
   while ((endIndex = _stringToBeSplitted.find(_delimeter, startIndex)) < _stringToBeSplitted.size())
   {
 	std::string val = _stringToBeSplitted.substr(startIndex, endIndex - startIndex);
@@ -117,6 +115,7 @@ std::vector<std::string> DB::splitString(std::string _stringToBeSplitted, std::s
   }
   return splittedString;
 }
+
 //############################################################################################################
 std::string DB::getFilenameOfContact(std::string _puplicKeyOfContactPerson)
 {
@@ -125,8 +124,6 @@ std::string DB::getFilenameOfContact(std::string _puplicKeyOfContactPerson)
   std::size_t hash = hasher(_puplicKeyOfContactPerson);
 
   std::string result = m_path.c_str() + std::to_string(hash) + ".db";
-
-
 
   return result;
 }

--- a/db/db.h
+++ b/db/db.h
@@ -6,7 +6,8 @@
 #include <vector>
 #include <fstream>
 #include <sstream>
-#include "dirent.h"
+
+typedef unsigned long long UINT64;
 
 class DB{
 public:
@@ -20,22 +21,21 @@ public:
   // Return: vector of all encounters. Each entry containing a tuple of 
   //   puplicKeyContactPerson	std::string
   //   puplicKeyInfectedPerson	std::string
-  //   coordinates longitude	double
-  //   coordinate altitude		double
+  //   coordinate latitude		double
+  //   coordinate longitude		double
   //   timestamp [ms]			UINT64
-  std::vector<std::tuple<std::string,std::string,double,double, UINT64>>  getEncounters(std::string puplicKeyContactPerson);
+  std::vector<std::tuple<std::string,std::string,double,double, UINT64>>  getEncounters(std::string _puplicKeyContactPerson);
 
-  // getEncounters
+  // submitEncounters
   // Submit all encounters of an infected person
   // Input:
   //  vector of all encounters. Each entry containing a tuple of 
   //   puplicKeyContactPerson	std::string
   //   puplicKeyInfectedPerson	std::string
-  //   coordinates longitude	double
-  //   coordinate altitude		double
+  //   coordinate latitude		double
+  //   coordinate longitude		double
   //   timestamp [ms]			UINT64
-  void submitEncounters(std::vector<std::tuple<std::string, std::string, double, double, UINT64>>);
-
+  void submitEncounters(std::vector<std::tuple<std::string, std::string, double, double, UINT64>> _encounters);
 
 private:
   std::vector<std::string> splitString(std::string _stringToBeSplitted, std::string _delimeter);

